build/common.sh

```diff
@@ -351,8 +351,11 @@ function kube::build::destroy_container() {
 #    local -r version_major="${BASH_REMATCH[1]}"
 #    local -r version_minor="${BASH_REMATCH[2]}"
 #    local -r version_patch="${BASH_REMATCH[3]}"
+#    local -r version_extra="${BASH_REMATCH[4]}"
+#    local -r version_prerelease="${BASH_REMATCH[5]}"
+#    local -r version_prerelease_rev="${BASH_REMATCH[6]}"
 function kube::release::parse_and_validate_release_version() {
-  local -r version_regex="^v(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)$"
+  local -r version_regex="^v(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)(-(beta|alpha)\\.(0|[1-9][0-9]*))?$"
   local -r version="${1-}"
   [[ "${version}" =~ ${version_regex} ]] || {
     kube::log::error "Invalid release version: '${version}'"
```

Comment:
I would suggest outputting the regex that was not matched here, to make the error message more informative.  Debugging without that will involve wading through this code, unnecessarily.